src/main/java/algorithms/curated170/medium/fourkeyskeyboard/FourKeysKeyboardDP.java

```diff
@@ -0,0 +1,27 @@
+package algorithms.curated170.medium.fourkeyskeyboard;
+
+public class FourKeysKeyboardDP {
+
+    public int maxA(int N) {
+        int[] best = new int[N + 1];
+
+        for (int press = 1; press <= N; press++) {
+            best[press] = best[press - 1] + 1;
+            if (press > 6) {
+                best[press] = Math.max(best[press], best[press - 6] * 5);
+                best[press] = Math.max(best[press], best[press - 5] * 4);
+                best[press] = Math.max(best[press], best[press - 4] * 3);
```

Comment:
You can choose:
```suggestion
               best[press] = Math.max(best[press], best[press - 3] * 2);
                best[press] = Math.max(best[press], best[press - 4] * 3);
                best[press] = Math.max(best[press], best[press - 5] * 4);
```

It is more relatable for me. Because it represents the selection between those choices more cleaner:
after f[i - 3], by ACP total length : f[i - 3] * 2
after f[i - 4], by ACP + P total length : f[i - 4] * 3
after f[i - 5], by ACP + PP total length : f[i - 5] * 4

Reply:
It should be a multiplication by 3, 4 and 5. Otherwise, I'm changing then.

Reply:
Added.

Reply:
It should also work like that:
```java
               best[press] = Math.max(best[press], best[press - 3] * 2);
                best[press] = Math.max(best[press], best[press - 4] * 3);
                best[press] = Math.max(best[press], best[press - 5] * 4);
```

But no problem. This is fine. I approve.

Reply:
Perhaps we never multiply by 5. But theoretically, we may be.

Reply:
Maybe there is a test-case we need. But according to LeetCode, we don't need it.

![image](https://user-images.githubusercontent.com/60903744/120936061-8cc32180-c70e-11eb-8fd7-d856b24bcac0.png)